source/extensions/filters/http/ext_authz/ext_authz.cc

```diff
@@ -5,11 +5,22 @@
 #include "common/http/codes.h"
 #include "common/router/config_impl.h"
 
+#include "extensions/filters/http/well_known_names.h"
+
 namespace Envoy {
 namespace Extensions {
 namespace HttpFilters {
 namespace ExtAuthz {
 
+void FilterConfigPerRoute::merge(const FilterConfigPerRoute& other) {
+  disabled_ = other.disabled_;
+  auto begin_it = other.context_extensions_.begin();
+  auto end_it = other.context_extensions_.end();
+  for (auto it = begin_it; it != end_it; ++it) {
+    context_extensions_[it->first] = it->second;
```

Comment:
Can you use some standard iterator thing for this, e.g. https://en.cppreference.com/w/cpp/container/unordered_map/merge?

Reply:
context_extensions_ is Protobuf::Map<ProtobufTypes::String, ProtobufTypes::String> 
the reason i don't use std::map is because i eventually move it to the check request, so I save copies doing it this way.

Initially I tried to use the `insert()` method, but that method doesn't override existing keys.

Reply:
Why stick with a proto map here? If you're going to copy, might as well go to `std` data structure that behaves more regularly?

Reply:
OK, but please comment this kind of optimization. IMHO, this copy elision doesn't really pay off when we're doing an RPC anyway in addition to each request, but maybe in the future when we're caching leases for authorization it would be good to have.

Also, need to merge master to resolve conflict.

Reply:
I tried to merge earlier and for some reason got a bunch of unrelated commits showed up. I figured I'd wait till we are ready to merge and rebase/force push

Reply:
Will add comment